fdbserver/KeyValueStoreSQLite.actor.cpp

```diff
@@ -435,8 +435,7 @@ Value encodeKVFragment( KeyValueRef kv, uint32_t index) {
 	}
 	// An increment is required if the high bit of the N-byte index value is set, since it is
 	// positive number but SQLite only stores signed values and would interpret it as negative.
-	if(index >> (8 * indexCode - 1))
-		++indexCode;
+	if (indexCode > 0 && index >> (8 * indexCode - 1)) ++indexCode;
```

Comment:
This seems right to me, but @satherton should be able to confirm.

Reply:
It's not wrong, but the check of indexCode > 0 is not actually necessary.  IndexCode starts at 0, and it is only incremented (at most 4 times), so it can only be positive.

I don't know what warning is being avoided here but does simply changing indexCode from int8_t to uint8_t achieve the same goal?  The way indexCode is used it should really be a uint8_t anyway.

Reply:
I think the problem being fixed is that if `indexCode` is 0, you'll be shifting by a negative number.

Reply:
Ahh okay, that makes sense, as right shift by a negative does not.  Since index == 0 == indexCode is the common case, it might be better to have lines 431 through 439 in an `if(index != 0)` block.